apps/server/src/modules/provisioning/service/provisioning.service.spec.ts

```diff
@@ -56,76 +69,128 @@ describe('ProvisioningService', () => {
+			provisioningStrategy: SystemProvisioningStrategy.SANIS,
+		});
+		const oauthDataDto: OauthDataDto = new OauthDataDto({
+			system: provisioningSystemDto,
+			externalUser: new ExternalUserDto({
+				externalId: 'externalUserId',
+			}),
 		});
-		const oidcStrategySystem: SystemDto = new SystemDto({
-			type: 'oidc',
-			provisioningStrategy: SystemProvisioningStrategy.OIDC,
+		const provisioningDto: ProvisioningDto = new ProvisioningDto({
+			externalUserId: 'externalUserId',
 		});
 
-		it('should throw error when system does not exists', async () => {
-			systemService.findById.mockRejectedValue(NotFoundException);
+		return {
+			systemId,
+			system,
+			provisioningSystemDto,
+			oauthDataDto,
+			provisioningDto,
+		};
+	};
+
+	describe('getData is called', () => {
+		const setup = () => {
+			const { systemId, system, provisioningSystemDto, oauthDataDto } = setupSystemData();
+			const accessToken = 'accessToken';
+			const idToken = 'idToken';
+
+			systemService.findById.mockResolvedValue(system);
+			provisioningStrategy.getData.mockResolvedValue(oauthDataDto);
+
+			return {
+				accessToken,
+				idToken,
+				systemId,
+				system,
+				provisioningSystemDto,
+				oauthDataDto,
+			};
+		};
+
+		describe('when the provisioning strategy is found', () => {
+			it('should call strategy.getData', async () => {
+				const { accessToken, idToken, systemId, provisioningSystemDto } = setup();
+
+				await service.getData(accessToken, idToken, systemId);
+
+				expect(provisioningStrategy.getData).toHaveBeenCalledWith(
+					new OauthDataStrategyInputDto({
+						accessToken,
+						idToken,
+						system: provisioningSystemDto,
+					})
+				);
+			});
+
+			it('should return the oauth data', async () => {
+				const { accessToken, idToken, systemId, oauthDataDto } = setup();
 
-			const process = () => provisioningService.process('accessToken', 'idToken', 'no system found');
+				const result: OauthDataDto = await service.getData(accessToken, idToken, systemId);
 
-			await expect(process()).rejects.toThrow(NotFoundException);
+				expect(result).toEqual(oauthDataDto);
+			});
 		});
 
-		it('should apply sanis provisioning strategy', async () => {
-			systemService.findById.mockResolvedValue(sanisStrategySystem);
+		describe('when no provisioning strategy is found', () => {
+			it('should throw an InternalServerErrorException', async () => {
+				const { accessToken, idToken } = setup();
+				const systemWithoutStrategy: SystemDto = new SystemDto({
+					type: '',
+					provisioningStrategy: SystemProvisioningStrategy.UNDEFINED,
+				});
+
+				systemService.findById.mockResolvedValue(systemWithoutStrategy);
 
-			await provisioningService.process('accessToken', 'idToken', sanisSystemStrategyId);
+				const promise: Promise<OauthDataDto> = service.getData(accessToken, idToken, 'systemId');
 
-			expect(sanisProvisioningStrategy.apply).toHaveBeenCalled();
+				await expect(promise).rejects.toThrow(InternalServerErrorException);
```

Comment:
If you throw more than one InternalServerErrorException mabye you should expect also an error message just to be sure to get the right exception